### ISSUE TITLE
Workout App

### ISSUE THREAD

**Issue by conroy-ricketts** [KnightHacks/projects-proposals-2021#4]
### What is your full name?

Conroy Ricketts 

### What is your Knights email?

conroy.ricketts@knights.ucf.edu 

### Are you interested in presenting your idea at the interest meeting?

Yes

### What's your idea about?

A mobile app that allows users to create a workout template, add exercises, add a certain number of sets and reps (or a time duration), utilize a built-in rest timer, and see a history of their progress over time. We could also implement a set of sample workout templates or exercise instructions.

**Comment by ElijahMSmith** [comment]
This is an excellent idea! I think it's a very reasonable proposition for a single semester and something that's very useful to a wide variety of people.

A few questions and suggestions:
Is there any specific information you want to keep in the history log? It would definitely have to have all the workouts you did and the time for each/the whole set, but I think it would be cool if there were also a long-term breakdown that automatically shows your progress over time at a particular workout set so that the user doesn't have to scroll through and personally evaluate, unless they want to.

I also think it would be cool to share your progress with other people! Obviously setting up an entire network/social feature is beyond one semester of work, but it might be cool to be able to export/share a single workout result or your progress for sharing on whatever social platform you use.

**Comment by conroy-ricketts** [comment]
Yeah, so the history log would show how many reps (or the time duration) and sets you did for each excerice of your previous workouts. The history log can also mark your personal records and keep a record of the time it took to complete each workout. I like your idea and it could definitely work with the app! Maybe there could be a visually nice looking line graph for certain excerises that the user chooses and it can show how their total volume increases over time. And yeah, a feature to share a snapshot of one of the history logs would be nice too!
